guac_core/src/channel_client/channel_manager.rs

```diff
@@ -242,27 +293,29 @@ impl ChannelManager {
 
     pub fn create_payment(&mut self) -> Result<UpdateTx, Error> {
         match self {
-            ChannelManager::Open { ref mut state } | ChannelManager::Joined { ref mut state } => {
-                Ok(state.create_payment()?)
-            }
+            ChannelManager::Open { ref mut state }
+            | ChannelManager::Joined { ref mut state }
+            | ChannelManager::PendingJoin { ref mut state, .. } => Ok(state.create_payment()?),
             // TODO: Handle close and dispute
             _ => bail!("we can only create payments in open or joined"),
         }
     }
 
-    pub fn rec_payment(&mut self, payment: UpdateTx) -> Result<UpdateTx, Error> {
+    pub fn rec_payment(&mut self, payment: &UpdateTx) -> Result<UpdateTx, Error> {
```

Comment:
Come on, why abbreviate the function name? Why do that and at the same time allow for names like `ChannelManagerAction` or `SendChannelProposal`? I say they get called `receive_*` instead